articles/dms/howto-mysql-to-azure-mysql-powershell.md

```diff
@@ -0,0 +1,657 @@
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "WriteDataRangeBatchTaskCount" $WriteDataRangeBatchTaskCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "MaxBatchCacheSizeMb" $MaxBatchCacheSizeMb;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "ThrottleQueryTableDataRangeTaskAtAvailableMemoryMb" $ThrottleQueryTableDataRangeTaskAtAvailableMemoryMb;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "ThrottleQueryTableDataRangeTaskAtBatchCount" $ThrottleQueryTableDataRangeTaskAtBatchCount;
+AddOptionalSetting $offlineMigTaskProperties.input.optionalAgentSettings "DelayProgressUpdatesInStorageInterval" $DelayProgressUpdatesInStorageInterval;
+```
+
+## Step 11: Running the migration task
+
+After configuration of the task, you will now execute the task and wait for the migration to complete.
+
+The following script invokes the configured migration task and waits for it to complete.
+
+```powershell
+# Running the migration scenario
+[string] $TaskName = "mysqlofflinemigrate"
+
+LogMessage -Message "Running data migration scenario ..." -IsProcessing $true
+$summary = @{
+    "SourceServer"   = $SourceServerName;
+    "SourceDatabase" = $SourceDatabaseName;
+    "TargetServer"   = $TargetServerName;
+    "TargetDatabase" = $TargetDatabaseName;
+    "TableCount"     = $tableMap.Count;
+    "StartedOn"      = $offlineMigTaskProperties.input.startedOn;
+}
+
+Write-Host "Job Summary:" -ForegroundColor Yellow
+Write-Host $(ConvertTo-Json $summary) -ForegroundColor Yellow
+
+$migrationResult = RunScenario -MigrationService $dmsService `
+    -MigrationProject $dmsProject `
+    -ScenarioTaskName $TaskName `
+    -TaskProperties $offlineMigTaskProperties
+
+LogMessage -Message "Migration completed with status - $($migrationRsult.state)"
+#Checking for any errors or warnings captured by the task during migration
+$dbLevelResult = $migrationResult.output | Where-Object { $_.resultType -eq "DatabaseLevelOutput" } 
+$migrationLevelResult = $migrationResult.output | Where-Object { $_.resultType -eq "MigrationLevelOutput" }
+if ($dbLevelResult.exceptionsAndWarnings) {
+    Write-Host "Following database errors were captured: $($dbLevelResult.exceptionsAndWarnings)" -ForegroundColor Red
+}
+if ($migrationLevelResult.exceptionsAndWarnings) {
+    Write-Host "Following migration errors were captured: $($migrationLevelResult.exceptionsAndWarnings)" -ForegroundColor Red
+}
+```
+
+## Step 12: Deleting the Database Migration Service
+
+If you're not going to continue to use the Database Migration Service, then you can delete the service using the [Remove-AzDataMigrationService](/powershell/module/az.datamigration/remove-azdatamigrationservice?) command.
+
+The following script deletes the Azure Database Migration Service instance and its associated projects.
+
+```powershell
+Remove-AzDataMigrationService -ResourceId $($dmsService.ResourceId)
```

Comment:
Since creating and removing a DMS resource is expensive, we should probably have these lines commented out by default.

Reply:
In the documentation it is written in a different section so modified the section description:
"The same Database Migration Service can be used for multiple migrations so the instance once created can be re-used. If you're not going to continue to use the Database Migration Service, then you can delete the service using the [Remove-AzDataMigrationService](/powershell/module/az.datamigration/remove-azdatamigrationservice?) command."

The complete end to end script does not even has the delete command but it is not yet hosted.